modules/n1ql/pages/n1ql-language-reference/with-recursive.adoc

```diff
@@ -212,8 +212,140 @@ SELECT * FROM rcte;
 ----
 ====
 
+.Examples handling hierarchical or tree-structured data
+====
+Before trying out these examples, you must:
+
+. Create a dataset containing employee information:
++
+[source,sqlpp]
+----
+INSERT INTO employees
+   VALUES ("e1", {
+       "employee_id": 1,
+       "employee_name": "John"
+       }),
+   VALUES ("e2", {
+       "employee_id": 2,
+       "employee_name": "Emily",
+       "manager_id": 1
+       }),
+   VALUES ("e3", {
+       "employee_id": 3,
+       "employee_name": "Mike",
+       "manager_id": 1
+       }),
+   VALUES ("e4", {
+       "employee_id": 4,
+       "employee_name": "Sarah",
+       "manager_id": 2
+       }),
+   VALUES ("e5", {
+       "employee_id": 5,
+       "employee_name": "Alex",
+       "manager_id": 3
+       }),
+   VALUES ("e6", {
+       "employee_id": 6,
+       "employee_name": "Lisa",
+       "manager_id": 2
+       })
+RETURNING *;
+----
++
+. Create a new index:
++
+[source,sqlpp]
+----
+CREATE INDEX e_idx ON employees(manager_id INCLUDE MISSING, employee_id, employee_name);
+----
+
+.Example finding the level of each employee in the organization
+[source,sqlpp]
+----
+WITH recursive emplLevel AS (
+   SELECT e.employee_id, e.employee_name, 0 lvl
+   FROM employees e
+   WHERE manager_id IS MISSING
+       UNION
+   SELECT e1.employee_id, e1.employee_name, e1.manager_id, emplLevel.lvl+1
+   lvl
+   FROM employees e1 JOIN emplLevel
+   ON e1.manager_id = emplLevel.employee_id
+)
+SELECT * FROM emplLevel;
+----
+
+.Example resulting in the reporting hierarchy of the employees in the organization
+[source,sqlpp]
+----
+SELECT e.employee_id, e.employee_name, e.manager_id,
+(
+    WITH RECURSIVE cte AS (
+        SELECT e1.employee_id, e1.employee_name, e1.manager_id
+        FROM employees e1 WHERE e1.manager_id = e.employee_id
+            UNION
+        SELECT e2.employee_id, e2.employee_name, e2.manager_id
+        FROM employees e2, cte
+        WHERE e2.manager_id = cte.employee_id
+    )
+    SELECT cte.* FROM cte
+) as reportsTo
+FROM employees e;
+----
```

Comment:
Show the results. (Feel free to cut the results down with an ellipsis `// ...` if you feel they're too long, as long as the different reporting structures are clear.)

```suggestion
----

.Results
[source,json]
----
[
  {
    "employee_id": 1,
    "employee_name": "John",
    "reportsTo": [
      {
        "employee_id": 2,
        "employee_name": "Emily",
        "manager_id": 1
      },
      {
        "employee_id": 3,
        "employee_name": "Mike",
        "manager_id": 1
      },
      {
        "employee_id": 4,
        "employee_name": "Sarah",
        "manager_id": 2
      },
      {
        "employee_id": 6,
        "employee_name": "Lisa",
        "manager_id": 2
      },
      {
        "employee_id": 5,
        "employee_name": "Alex",
        "manager_id": 3
      }
    ]
  },
  {
    "employee_id": 2,
    "employee_name": "Emily",
    "manager_id": 1,
    "reportsTo": [
      {
        "employee_id": 4,
        "employee_name": "Sarah",
        "manager_id": 2
      },
      {
        "employee_id": 6,
        "employee_name": "Lisa",
        "manager_id": 2
      }
    ]
  },
  {
    "employee_id": 3,
    "employee_name": "Mike",
    "manager_id": 1,
    "reportsTo": [
      {
        "employee_id": 5,
        "employee_name": "Alex",
        "manager_id": 3
      }
    ]
  },
  {
    "employee_id": 4,
    "employee_name": "Sarah",
    "manager_id": 2,
    "reportsTo": []
  },
  {
    "employee_id": 6,
    "employee_name": "Lisa",
    "manager_id": 2,
    "reportsTo": []
  },
  {
    "employee_id": 5,
    "employee_name": "Alex",
    "manager_id": 3,
    "reportsTo": []
  }
]
----
```